sys/net/gnrc/link_layer/gnrc_mac/timeout.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2017 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc_mac
+ * @{
+ *
+ * @file
+ * @brief       Implementation of timeout module of GNRC_MAC
+ *
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ * @}
+ */
+
+#include "net/gnrc.h"
+#include "net/gnrc/mac/timeout.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+void gnrc_mac_init_timeouts(gnrc_mac_timeout_t *mac_timeout,
+                            gnrc_mac_timeout_event_t timeouts[],
+                            uint8_t num)
+{
+    assert(mac_timeout);
+    assert(timeouts);
+    assert(num);
+
+    mac_timeout->timeouts = timeouts;
+    mac_timeout->timeout_num = num;
+
+    for (int i = 0; i < mac_timeout->timeout_num; i++) {
+        mac_timeout->timeouts[i].msg_event.event.next = NULL;
+        mac_timeout->timeouts[i].type = GNRC_MAC_TIMEOUT_DISABLED;
+    }
+
+    evtimer_init_msg(&mac_timeout->evtimer);
+}
+
+/* Return index >= 0 if found timeout, -ENONENT if not found */
+static int _gnrc_mac_find_timeout(gnrc_mac_timeout_t *mac_timeout, gnrc_mac_timeout_type_t type)
+{
+    assert(mac_timeout);
+    assert(mac_timeout->timeout_num);
+
+    for (unsigned i = 0; i < mac_timeout->timeout_num; i++) {
+        if (mac_timeout->timeouts[i].type == type) {
+            return i;
+        }
+    }
+    return -ENOENT;
+}
+
+inline bool gnrc_mac_timeout_is_running(gnrc_mac_timeout_t *mac_timeout,
```

Comment:
This needs to be `static inline`.

```suggestion
static inline bool gnrc_mac_timeout_is_running(gnrc_mac_timeout_t *mac_timeout,
```

This is what LLVM is complaining about.